plugins/out_stackdriver/stackdriver.c

```diff
@@ -2510,6 +2539,177 @@ static void update_retry_metric(struct flb_stackdriver *ctx,
 }
 #endif
 
+#define GRPC_STATUS_CODES_SIZE 32
+#define PARTIAL_SUCCESS_GRPC_TYPE "type.googleapis.com/google.logging.v2.WriteLogEntriesPartialErrors"
+#define PARTIAL_SUCCESS_GRPC_TYPE_SIZE 66
+
+static int parse_partial_success_response(struct flb_http_client* c,
+                                          struct flb_stackdriver* ctx,
+                                          uint64_t ts,
+                                          int total_events,
+                                          int* grpc_status_codes) {
+    int ret;
+    int root_type;
+    int i;
+    int log_entry_ret;
+    int code_ret;
+    char* buffer;
+    size_t size;
+    size_t off = 0;
+    msgpack_unpacked result;
+    msgpack_object root;
+    msgpack_object error_map;
+    msgpack_object details_arr;
+    msgpack_object details_map;
+    msgpack_object logEntryErrors_map;
+    msgpack_object logEntryError_key;
+    msgpack_object logEntryError_map;
+    msgpack_object logEntryCode;
+    msgpack_object at_type;
+
+    if (c->resp.status != 400 && c->resp.status != 403) {
+        return -1;
+    }
+
+    ret = flb_pack_json(c->resp.payload, c->resp.payload_size,
+                        &buffer, &size, &root_type, NULL);
+    if (ret != 0) {
+        flb_plg_error(ctx->ins, "failed to parse json into msgpack\n%s",
+                      c->resp.payload);
+        return -1;
+    }
+
+    msgpack_unpacked_init(&result);
+    ret = msgpack_unpack_next(&result, buffer, size, &off);
+    if (ret != MSGPACK_UNPACK_SUCCESS) {
+        flb_plg_error(ctx->ins, "Cannot unpack %s response\n%s",
```

Comment:
We shouldn't put `\n` in the error logs because we don't pick up multiline when we ingest these; best to keep them on the same line when we can